qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
     return ", ".join(shared)
 
 
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_single_study_info(study, info, study_proc, proc_samples):
+    """Clean up and add to the study info for HTML purposes
+
+    Parameters
+    ----------
+    study : Study object
+        The study to build information for
+    info : dict
+        Information from Study.get_info
+
+    Returns
+    -------
+    dict
+        info-information + extra information for the study,
+        slightly HTML formatted
+    """
+    PI = StudyPerson(info['principal_investigator_id'])
+    status = study.status
+    if info['pmid'] is not None:
+        info['pmid'] = ", ".join([pubmed_linkifier([p])
+                                  for p in info['pmid']])
+    else:
+        info['pmid'] = ""
+    if info["number_samples_collected"] is None:
+        info["number_samples_collected"] = 0
+    info["shared"] = _get_shared_links_for_study(study)
+    info["num_raw_data"] = len(study.raw_data())
+    info["status"] = status
+    info["study_id"] = study.id
+    info["pi"] = study_person_linkifier((PI.email, PI.name))
+    del info["principal_investigator_id"]
+    del info["email"]
+    # Build the proc data info list for the child row in datatable
+    info["proc_data_info"] = _build_single_proc_data_info(
+        study, study_proc, proc_samples)
+    return info
+
+
+def _build_single_proc_data_info(study, study_proc, proc_samples):
```

Comment:
But then this is not building the information of a single processed data. IMOO, it is not weird. When designing this kind of operations, you try to design them in a way that the smallest chunk of work is created. Also, I don't get your point. The break up that I'm proposing is the same that the one for the study. Instead of building the entire study list, you call a function for each study. Here, you call a function for each processed data. Line 79 will change to something like:
```
info["proc_data_info"] = [_build_single_proc_data_info(pd_id, proc_samples[pd_id]) for pd_id in study_proc[study.id]]
```